JetInstance.newJetClient is not compatible with new async connection mode

It's not possible to create a new Jet client using the [async connection mode](http://docs.hazelcast.org/docs/latest-dev/manual/html-single/index.html#async-start-and-reconnect-modes), as you get an exception:

```
Caused by: com.hazelcast.client.HazelcastClientOfflineException: Client connecting to cluster
	at com.hazelcast.client.spi.ProxyManager.findNextAddressToSendCreateRequest(ProxyManager.java:443) ~[hazelcast-client-3.10.jar:3.10]
	at com.hazelcast.client.spi.ProxyManager.initialize(ProxyManager.java:430) ~[hazelcast-client-3.10.jar:3.10]
	at com.hazelcast.client.spi.ProxyManager.initializeWithRetry(ProxyManager.java:395) ~[hazelcast-client-3.10.jar:3.10]
	at com.hazelcast.client.spi.ProxyManager.getOrCreateProxy(ProxyManager.java:316) ~[hazelcast-client-3.10.jar:3.10]
	at com.hazelcast.client.impl.HazelcastClientInstanceImpl.getDistributedObject(HazelcastClientInstanceImpl.java:707) ~[hazelcast-client-3.10.jar:3.10]
	at com.hazelcast.client.impl.HazelcastClientInstanceImpl.getMap(HazelcastClientInstanceImpl.java:511) ~[hazelcast-client-3.10.jar:3.10]
	at com.hazelcast.jet.impl.JobRepository.<init>(JobRepository.java:115) ~[classes/:na]
	at com.hazelcast.jet.impl.AbstractJetInstance.<init>(AbstractJetInstance.java:39) ~[classes/:na]
	at com.hazelcast.jet.impl.JetClientInstanceImpl.<init>(JetClientInstanceImpl.java:67) ~[classes/:na]
	at com.hazelcast.jet.Jet.getJetClientInstance(Jet.java:99) ~[classes/:na]
	at com.hazelcast.jet.Jet.newJetClient(Jet.java:87) ~[classes/:na]
```

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJetInstance.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJetInstance.java
@@ -23,20 +23,22 @@ import com.hazelcast.jet.config.JobConfig;
 import com.hazelcast.jet.IListJet;
 import com.hazelcast.jet.IMapJet;
 import com.hazelcast.jet.JetCacheManager;
+import com.hazelcast.jet.impl.util.Util;
 import com.hazelcast.jet.stream.impl.IListDecorator;
 import com.hazelcast.jet.stream.impl.IMapDecorator;
 
 import javax.annotation.Nonnull;
+import java.util.function.Supplier;
 
 abstract class AbstractJetInstance implements JetInstance {
     private final HazelcastInstance hazelcastInstance;
     private final JetCacheManagerImpl cacheManager;
-    private final JobRepository jobRepository;
+    private final Supplier<JobRepository> jobRepository;
 
     AbstractJetInstance(HazelcastInstance hazelcastInstance) {
         this.hazelcastInstance = hazelcastInstance;
         this.cacheManager = new JetCacheManagerImpl(this);
-        this.jobRepository = new JobRepository(this, null);
+        this.jobRepository = Util.memoizeConcurrent(() -> new JobRepository(this, null));
     }
 
     @Nonnull @Override
@@ -75,7 +77,7 @@ abstract class AbstractJetInstance implements JetInstance {
     }
 
     protected long uploadResourcesAndAssignId(JobConfig config) {
-        return jobRepository.uploadJobResources(config);
+        return jobRepository.get().uploadJobResources(config);
     }
 
 }

